File: c++/assign4/assign4.cpp
#include <iostream>
#include <iomanip>
#include <cmath>
using namespace std;

int main()
{
    //Get Input from user
    int month = 0;
    double balance, payment, APR, interestRate = 0;
    cout << "Enter balance: ";
    cin >> balance;
    cout << "Enter payment: ";
    cin >> payment;
    cout << "Enter APR: ";
    cin >> APR;

    //Compute interest and output result
    cout << "Month " << " Int." << setw(8) << "Pay \t" << "Balance \n";
    while (interestRate >= 0) {
        cout << fixed << setprecision(2) << month << "   " << setw(8) << interestRate << setw(8) << payment << "\t" << balance << endl;
        month++;
        interestRate = (APR / 100) / 12 * balance;
        balance = balance - payment + interestRate;
    }

    return 0;
}

File: JS/assign4.js
let balance = 1000;
let payment = 150;
let APR = 19.9, interestRate = 0, month = 0;

// Compute output
console.log('Month  ' + ' int    ' + ' Pay ' + ' balance ');

while (interestRate >= 0) {
	console.log(month + '      ' + interestRate.toPrecision(3) + '    ' + payment + '    ' + balance.toPrecision(4));
	month++
	interestRate = (APR / 100) / 12 * balance;
	balance = balance - payment + interestRate;
}
